src/Shared/UnitTests/FileMatcher_Tests.cs

```diff
@@ -1040,8 +1040,15 @@ private void GetMatchingDirectories(string[] candidates, string path, string pat
                             int nextSlash = normalizedCandidate.IndexOfAny(FileMatcher.directorySeparatorCharacters, path.Length + 1);
                             if (nextSlash != -1)
                             {
-                                string match = normalizedCandidate.Substring(0, nextSlash + 1);
+                                string match;
+
+                                //UNC paths start with a \\ fragment. Match against \\ when path is empty (i.e., inside the current working directory)
+                                match = normalizedCandidate.StartsWith(@"\\") && string.IsNullOrEmpty(path)
+                                    ? @"\\"
+                                    : normalizedCandidate.Substring(0, nextSlash);
```

Comment:
It was transplanted from master. I'd keep it as is and not make the conditions more complicated.